File: docs/content/recipes/data-management/server-side-rails/server/order.rb
# app/models/order.rb
#
# Order model for the Order Management demo grid.
#
# Fields match the Handsontable column definitions:
#   id           - auto-increment primary key, used as rowId on the frontend
#   order_number - short identifier shown in the first column
#   customer     - customer display name
#   status       - workflow state (pending, paid, shipped, delivered, cancelled)
#   total        - order total; DecimalField avoids floating-point rounding
#   created_at   - read-only timestamp filled on insert
class Order < ApplicationRecord
  STATUSES = %w[pending paid shipped delivered cancelled].freeze

  validates :order_number, presence: true, uniqueness: true
  validates :customer,     presence: true
  validates :status,       inclusion: { in: STATUSES }
  validates :total,        numericality: { greater_than_or_equal_to: 0 }

  # Keeps the initial grid display predictable before the user sorts.
  default_scope -> { order(created_at: :desc) }
end

File: docs/content/recipes/data-management/server-side-aspnet/server/Order.cs
namespace OrdersApi.Models;

public class Order
{
    public int Id { get; set; }
    public string OrderNumber { get; set; } = string.Empty;
    public string Customer { get; set; } = string.Empty;
    public string Status { get; set; } = "pending";
    public decimal Total { get; set; }
    public DateTime CreatedAt { get; set; } = DateTime.UtcNow;
}
